app/controllers/movies_controller.rb

```diff
@@ -1,42 +1,12 @@
 class MoviesController < ApplicationController
-  helper_method :genres
-
   def index
-    conn = Faraday.new(url: 'https://api.themoviedb.org')
-    uri = params[:query].nil? ? 'movie/top_rated' : 'search/movie'
-
-    @movies = []
-    2.times do |n|
-      response = conn.get("/3/#{uri}") do |rec|
-        rec.params[:api_key] = ENV['TMDB_API_KEY']
-        rec.params[:query] = params[:query] if params[:query]
-        rec.params[:language] = 'en-US'
-        rec.params[:page] = 1 + n
-        rec.params[:include_adult] = false
-      end
-      @movies << JSON.parse(response.body, symbolize_names: true)[:results]
-    end
-    @movies.flatten!
+    @movies = MovieFacade.search(params[:query])
   end
 
   def show
-    conn = Faraday.new(url: 'https://api.themoviedb.org')
-    uri = "movie/#{params[:id]}"
-    response = conn.get("/3/#{uri}") do |rec|
-      rec.params[:api_key] = ENV['TMDB_API_KEY']
-      rec.params[:movie_id] = params[:id]
-      rec.params[:append_to_response] = 'credits,reviews'
-    end
-    @movie = JSON.parse(response.body, symbolize_names: true)
-    @cast = @movie[:credits][:cast][0..9]
-    @reviews = @movie[:reviews][:results]
-  end
-
-  private
-
-  def genres(genre_array)
-    genre_array.map do |genre|
-      genre[:name]
-    end.join(', ')
+    @movie = MovieFacade.details(params[:id])
+    # WTF should we do about this???
+    @cast = @movie.cast[0..9]
```

Comment:
Would it be better to just call `@movie.cast` and `@movie.reviews` in the view, rather than assigning them as instance variables?